File: 8a.php
<?php

$fname = "8.dat";
$fin = fopen($fname, "r");
$data = fread($fin, filesize($fname));
$lines = explode("\n", $data);
unset($lines[count($lines)-1]);
$total = 0;

foreach ($lines as $line) {
    $total += strlen($line);
    /* echo strlen($line)." ".$line." - "; */
    $line = substr($line, 1, strlen($line)-2);
    if (preg_match_all("/\\\\x[0-9a-f]{2}/", $line, $matches)) {
        /* print_r($matches); */
        foreach ($matches[0] as $m) {
            $line =  str_replace($m, chr(hexdec($m)), $line);
        }
    }
    $line = str_replace("\\\\", "\\", $line);
    $line = str_replace("\\\"", "\"", $line);
    $total -= strlen($line);
    /* echo $line.PHP_EOL; */
}
echo $total.PHP_EOL;

fclose($fin);


?>

File: 8a.py
#!/usr/bin/python3
import re

with open('8.dat') as f:
    data = f.read()
lines = data.split('\n')
lines.pop()
total = 0
for line in lines:
    tr = line[1:-1]
    matches = re.findall(r'\\x[0-9a-f]{2}', line)
    for m in matches:
        tr = tr.replace(m, chr(int("0"+m[1:4], 16)))
    tr = tr.replace("\\\\", "\\")
    tr = tr.replace("\\\"", "\"")
    # print(len(line), len(tr), line)
    total += len(line) - len(tr)
print(total)
